src/Service/NuRaftStateMachine.cpp

```
@@ -75,9 +76,9 @@ NuRaftStateMachine::NuRaftStateMachine(
     if (last_snapshot != nullptr)
         applySnapshotImpl(*last_snapshot);
 
-    task_manager = cs_new<RaftTaskManager>(snapshot_dir);
+    committed_log_manager = cs_new<LastCommittedIndexManager>(log_dir);
     uint64_t previous_last_commit_id = 0; /// Last committed idx of the previous startup, we should apply log to here.
-    task_manager->getLastCommitted(previous_last_commit_id);
+    committed_log_manager->get(previous_last_commit_id);
 
     LOG_INFO(log, "Loading logs from {} to {}", last_committed_idx + 1, previous_last_commit_id);
     replayLogs(log_store_, last_committed_idx + 1, previous_last_commit_id);
```

Maybe throw an exception, if previous_last_commit_id < last_committed_idx.